### PR TITLE
Task order validations

### PR DESCRIPTION
**Pull request by drummond-dds**
This PR implements the ["Save as Draft" feature](https://www.pivotaltracker.com/story/show/162772188).

When clicking "Save as Draft" we set a hidden field via the vue component called `is_draft` to `True`. This field is only an attribute and it does not get saved to the database. When running the `validate()` on a form when the `is_draft` attribute is set to `True` we run through the validations we run the first validation (`OptionalAsDraft()`) which returns a status as `valid=True` for that field and halts all further validations down the chain.

**NOTE**: `OptionalAsDraft()` must be the first validation to run.

When we want to check the completion status we run the method `section_completion_status()` which replaces `is_section_complete()` and this method now returns the form steps completion state of either `["complete", "draft", "incomplete"]`.

The form steps component now uses the `section_completion_status` to display either complete, draft, incomplete in the timeline at the top.

The method `all_sections_complete()` has also been changed to make use of `section_completion_status()`.

### CODE REVIEW

**Comment by patricksmithdds** [comment]
@drummond-dds Do you mind updating the description with a summary of how this feature is intended to work?

File: atst/domain/task_orders.py [hunk 1]
```diff
@@ -89,21 +60,42 @@ def update(cls, user, task_order, **kwargs):
         return task_order
 
     @classmethod
-    def is_section_complete(cls, task_order, section):
-        if section in TaskOrders.SECTIONS:
-            for attr in TaskOrders.SECTIONS[section]:
-                if getattr(task_order, attr) is None:
-                    return False
+    def section_completion_status(cls, task_order, section):
```
Comment (drummond-dds) [inline atst/domain/task_orders.py]: Move this to the routes file

File: atst/forms/task_order.py [hunk 2]
```diff
@@ -65,35 +82,52 @@ class AppInfoForm(CacheableForm):
...
         translate("forms.task_order.team_experience.label"),
         description=translate("forms.task_order.team_experience.description"),
         choices=TEAM_EXPERIENCE,
         default="",
+        validators=[OptionalAsDraft()],
     )
 
 
 class FundingForm(CacheableForm):
+    is_draft = BooleanField("Is draft", default=False)
```
Comment (drummond-dds) [inline atst/forms/task_order.py]: Experiment with adding a separate draft route and not having the `is_draft` attribute

File: js/components/savedraft.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+export default {
```
Comment (drummond-dds) [inline js/components/savedraft.js]: Look at button action attributes on form to allow submitting to two routes from two buttons from one form

File: atst/forms/task_order.py [hunk 4]
```diff
@@ -26,28 +26,44 @@
...
+    def _requiredUnlessDraft(form, field):
+        if form.is_draft.data:
+            field.errors[:] = []
+            raise StopValidation()
+
+    return _requiredUnlessDraft
+
+
 class AppInfoForm(CacheableForm):
+    is_draft = BooleanField("Is draft", default=False)
```
Comment (drummond-dds) [inline atst/forms/task_order.py]: Look into moving draft validation logic into `CacheableForm` or a subclass of that

File: atst/forms/task_order.py [hunk 5]
```diff
@@ -26,28 +26,44 @@
...
     portfolio_name = StringField(
         translate("forms.task_order.portfolio_name_label"),
         description=translate("forms.task_order.portfolio_name_description"),
+        validators=[OptionalAsDraft(), Required()],
```
Comment (drummond-dds) [inline atst/forms/task_order.py]: This should always be required

**Comment by drummond-dds** [comment]
Talk to UX about how to handle "Send invitation" when saving as draft

**Comment by drummond-dds** [comment]
Closing this PR for now

**Pull request closed without merging**